### PR TITLE
Adds an AI controllable powerloader drone

### PR DESCRIPTION
**Pull request by RipGrayson**
## About The Pull Request

Title.
The drone is intended to allow the AI to manage req or deliver supplies if nobody else is around, I've taken the liberty of adding it to all ship maps and modified the existing powerloader sprite so that the AI version is unique looking.

It cannot load the OB cannon or service the CAS plane, the logic for that is tricky and would bloat the PR, maybe in the future.

## Why It's Good For The Game

This is something that I've been asked for a bunch of times. The AI can't handle moving crates or help manage the ship because it has no hands, with this PR it can at least man requisitions if asked and move supplies onto the Alamo.

## Changelog
:cl:
add: Added a new powerloader drone that the AI can control.
add: Added the new type of drone to all ship maps.
/:cl:

### CODE REVIEW

**Comment by FosterGrime-Heymynameslouie** [comment]
would making it be able to close and open the crate be hard? 
would be cool if AI can also vend from the vendors, vend a crate then close it for the stuff to go in the crate

or maybe repurpose trash bag code so that if you have a crate in hand it slurps up all the items on the ground and goes inside crate

cool pr please sepdemgherg instnatly

**Comment by RipGrayson** [comment]
> Isnt the point of AI that they have no hands

This is less hands and more of the ability to store and drop a single item in another area. I'm willing to nerf it as needed, perhaps I could make it only apply to crates?

**Comment by zzurge** [comment]
Out fucking standing,Now i dont have to ebeg some poor rando to use the starting conveyors to act as ARO AI

**Comment by Lumipharon** [comment]
Not a huge fan of giving AI the ability to directly interact with stuff like this.

**Comment by RipGrayson** [comment]
> Not a huge fan of giving AI the ability to directly interact with stuff like this.

Tivi said it was fine as long as it stayed hard locked to shipside, if you want I can lock it even further to req.

**Comment by Losenis** [comment]
Can it actually open/close crates or handle items or is it just a powerloader?

**Comment by RipGrayson** [comment]
> Can it actually open/close crates or handle items or is it just a powerloader?

As of current it has the extra ability to toggle crates and can handle items shipside, that may or may not stay. Lumi and Tivi aren't fond of the AI having "hands" so it might have to be just a crate hauler.

**Review by Kuro020 — APPROVED** [review]

File: code/modules/vehicles/unmanned/unmanned_droid.dm [hunk 1]
```diff
@@ -145,3 +145,63 @@
...
+	if(is_ground_level(z) && !isdropshiparea(get_area(src))) //AI powerloader is confined to shipside or the alamo
+		to_chat(user, "Connection too weak, return the droid shipside first.")
+		return
+	if(!COOLDOWN_CHECK(src, clamp_cooldown))
+		return
+	if(cargo && Adjacent(target) && istype(target, /obj/structure/closet))
+		var/obj/structure/closet/attackedcloset = clamptarget
+		attackedcloset.toggle()
+	else if(cargo)
+		to_chat(user, "You unload [cargo].")
```
Comment (TiviPlus) [inline code/modules/vehicles/unmanned/unmanned_droid.dm:182]: baloons

**Comment by FosterGrime-Heymynameslouie** [comment]
how do you close the crate as an AI?

**Comment by XSlayer300** [comment]
Add a help text tidbit for the controls?

**Comment by RipGrayson** [comment]
> Add a help text tidbit for the controls?

None of the other droids have helper text associated and they have the same complexity as this drone.

I can post it here if you want, click object to pick up, click anywhere to drop it. If you're holding an item and click a crate it will toggle it open or closed.

**Pull request merged**